packages/fast-usdc/src/exos/settler.js

```diff
@@ -151,17 +151,23 @@ export const prepareSettler = (
             return;
           }
 
-          if (!addressTools.hasQueryParams(tx.receiver)) {
+          let endUserDestination;
+          try {
+            const { EUD } = decodeAddressHook(tx.receiver).query;
+            if (!EUD) {
+              console.log('no EUD parameter', tx.receiver);
+              return;
+            }
+            if (!matches(EUD, M.string())) {
```

Comment:
no need to go outside JS here.

```suggestion
            if (typeof EUD !== 'string') {
```

A pattern match for the whole return value of `decodeAddressHook(...)` would make sense.
Does `@endo/patterns` have a thing between `matches()` and `mustMatch()`? i.e. if this doesn't match, give me a good reason why not, not just a boolean `false`.